### PR TITLE
IDE, AppStorage fixes - load improvements

### PR DESCRIPTION
**Pull request by gokmen**
This PR fixes most of the resurrection issues on IDE and it also introduces faster loads with AppStorage improvements. This is a set of required changes to make #9756 happen, I'll continue over this one.

ref: http://recordit.co/hW8hkBkYvX

Marathon provider is using `/root` as home folder based on the `klient.info` response:

### CODE REVIEW

**Comment by gokmen** [comment]
I've reverted https://github.com/koding/koding/pull/9587/commits/fe820824e965014002c1deacf19bb327eca80bbd since the main issue in AppStorage is fixed in this PR cc/ @hakankaradis

**Comment by gokmen** [comment]
I'm done with this one but please don't merge it even it's ok, there are some experimental changes that might affect some place else that I didn't tested and verified on my own. We can merge this in once it's ready after the upcoming deploy to production. cc/ @sinan @szkl

**Comment by rjeczalik** [comment]
Will we have backward compatibility with old klient.info responses? I'm thinking of old managed klient which do not have autoupdates enabled.

**Comment by gokmen** [comment]
@rjeczalik I covered the case of error here https://github.com/koding/koding/pull/9804/files#diff-52d8d249234cc4460894a6fb44d36fbbR172 but not checked `info` result, good catch let me update.

File: client/ide/lib/index.coffee
Comment (sinan) [inline client/ide/lib/index.coffee]: Seems like this method became too big. Let's separate `if/else` section into separate methods.

Comment (gokmen) [inline client/ide/lib/index.coffee]: it's becoming shorter actually it was https://github.com/koding/koding/blob/master/client/ide/lib/index.coffee#L550-L597 (47 lines) but now https://github.com/koding/koding/blob/d8c0fd4522a2e8223437930943602aa8f110ab69/client/ide/lib/index.coffee#L534-L575 (41 lines) :) but I can split it out of course.

**Review by sinan — APPROVED** [review]

**Pull request merged**
